File: 98-ValidateBinarySearchTree/98-ValidateBinarySearchTree.c
// Last updated: 11/12/2025, 4:59:07 AM
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool checkhelper(struct TreeNode* root, long minimum, long maximum)
{
    if(root==NULL) return true;
    if(root->val<=minimum || root->val>=maximum) return false;
    return checkhelper(root->left, minimum,root->val) && checkhelper(root->right,root->val,maximum);
}
bool isValidBST(struct TreeNode* root)
{
    if(root==NULL) return true;
    return checkhelper(root,LONG_MIN,LONG_MAX);
}

File: 98-ValidateBinarySearchTree/98-ValidateBinarySearchTree.py
# Last updated: 2/7/2026, 9:27:54 PM
1# Definition for a binary tree node.
2# class TreeNode:
3#     def __init__(self, val=0, left=None, right=None):
4#         self.val = val
5#         self.left = left
6#         self.right = right
7class Solution:
8    def isValidBST(self, root: Optional[TreeNode]) -> bool:
9        def dfs(node,low,high):
10            if not node:
11                return True
12            if not low<node.val<high:
13                return False
14            return dfs(node.left,low,node.val) and dfs(node.right,node.val,high)
15        return dfs(root, float("-inf"),float("inf"))